### PR TITLE
docs: minor updates for Egress Gateway

### PR DESCRIPTION
**Pull request by julianwiedmann**
Fully reflect https://github.com/cilium/cilium/pull/32828 in the docs.

### CODE REVIEW

**Comment by julianwiedmann** [comment]
@ysksuzuki maybe we should log/warn from the EGW manager if DNS proxy is not in transparent mode? Similar to https://github.com/cilium/cilium/pull/32828/commits/feafd6dfafe4f8865e3bcd78017ec4364a079cc8 :grimacing:.

**Review by qmonnet — APPROVED** [review]

File: Documentation/network/egress-gateway.rst [hunk 1]
```diff
@@ -60,6 +60,9 @@ Additionally, the enablement of the egress gateway feature requires that both
 BPF masquerading and the kube-proxy replacement are enabled, which may not be
 possible in all environments (due to, e.g., incompatible kernel versions).
 
+When pods are selected by a DNS policy **and** an Egress gateway policy, the
+DNS proxy needs to operate in transparent mode (``--dnsproxy-enable-transparent-mode=true``).
```
Comment (ysksuzuki) [inline Documentation/network/egress-gateway.rst]: Is the dnsproxy transparent mode really needed?

The DNS traffic flows from the Pod to core-dns or node-dns, right? Are there any cases where we need to encap and redirect DNS queries to an external DNS to the GW?

Comment (julianwiedmann) [reply Documentation/network/egress-gateway.rst]: @ysksuzuki of course, you're right - the DNS traffic should stay in-cluster, so for EGW it doesn't matter whether we see the original source IP. I must have gotten mixed up with the changes we needed for envoy. Thank you!

**Review by ysksuzuki — APPROVED** [review]

**Comment by ysksuzuki** [comment]
Thanks!

**Pull request merged**